### PR TITLE
fix types and manifest for writers

### LINKED ISSUE #70: The `python_name` for command `napari-foobar.writer` is duplicated in the commands list
```yaml
    commands:
      - id: napari-foobar.get_reader
        python_name: napari_foobar._reader:napari_get_reader
        title: Open data with napari FooBar
      - id: napari-foobar.writer
        python_name: napari_foobar._reader:napari_get_reader
        title: Save data with napari FooBar
  ```

### LINKED ISSUE #71: writer type signatures are wrong
```py
    def write_single_image(path: str, data: Any): ...
    # should be changed to
    def write_single_image(path: str, data: Any, meta: dict): ...
    ```
    and
    ```py
    def write_multiple(path: str, data: List[ArrayLike], layer_types: List[str]): ...
    # should be change to
    def write_multiple(path: str, List[FullLayerData]): ...
    ```
    The args are passed [here](https://github.com/tlambert03/npe2/blob/921f5f120aff16ddbdb1c2c7de4c236eb860fcc7/npe2/io_utils.py#L185)

### PR DESCRIPTION
**Pull request by nclack**
closes #71 
closes #70 

I also tried to audit a bit based on the comments in #69 re `ArrayLike` etc.

### CODE REVIEW

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 1]
```diff
@@ -7,15 +7,18 @@
 Replace code below according to your needs.
 """
 from __future__ import annotations
-from typing import List
-from numpy.typing import ArrayLike
+from typing import TYPE_CHECKING, List, Any, Sequence, Tuple, Union
 
+if TYPE_CHECKING:
+    DataType = Union[Any, Sequence[Any]]
```
Comment (nclack) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:13]: `DataType` is a bit excessive maybe. Just replace w `Any`?

Comment (tlambert03) [reply {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:13]: can do, i do think broader is better.
but fwiw, that _is_ basically the layerdata protocol we expect.  (structural type as opposed to literal type).  so, without saying `Union[numpy.ndarray, dask.Array, etc...]` it beings captures what a user can expect from that object

Comment (tlambert03) [reply {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:13]: oh, sorry, my brain was still in npe2 😂  ... where we have another thing called DataType don't listen to me :)

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 2]
```diff
@@ -7,15 +7,18 @@
...
+    FullLayerData = Tuple[DataType, dict, str]
 
-def write_single_image(path: str, data: ArrayLike):
+
+def write_single_image(path: str, data: Any, meta: dict):
     """Writes a single image layer"""
```
Comment (tlambert03) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:18]: are we gonna do the rough "takes X and is expected to do Y and return Z" later?

**Review by tlambert03 — APPROVED** [review]

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py (hunk 2 above)
Comment (nclack) [reply {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:18]: yes

**Pull request merged**